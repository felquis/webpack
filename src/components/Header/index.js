import React, { Component } from 'react'
import styles from './style.css'

class Header extends Component {
  constructor(props) {
    super(props)

    this.state = {
      title: this.props.title
    }

    this.handleClick = this.handleClick.bind(this)
  }

  handleClick () {
    this.setState({
      title: 'Another Title'
    })
  }

  render () {
    return (
      <header onClick={this.handleClick} className={styles.header}>
        <h1>{this.state.title}</h1>
      </header>
    )
  }
}

export default Header
