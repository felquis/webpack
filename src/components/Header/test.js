import React from 'react'
import Header from './index.js'
import renderer from 'react-test-renderer'


it('should be ok', () => {
  const component = renderer.create(<Header title="Hello World" />)
  let tree = component.toJSON()

  expect(tree).toMatchSnapshot()

  tree.props.onClick()

  tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
