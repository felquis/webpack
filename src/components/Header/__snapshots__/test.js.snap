exports[`test should be ok 1`] = `
<header
  className="header"
  onClick={[Function]}>
  <h1>
    Hello World
  </h1>
</header>
`;

exports[`test should be ok 2`] = `
<header
  className="header"
  onClick={[Function]}>
  <h1>
    Another Title
  </h1>
</header>
`;
